src/main/java/TextStatsUtils.java

```diff
@@ -0,0 +1,37 @@
+import java.io.PrintWriter;
+
+public class TextStatsUtils {
+
+    private TextStatsUtils() {
+    }
+
+    public static void countAllChars(String text, PrintWriter writer) {
+        writer.println("<i>" + text + "</i>");
+        writer.println("<h1>Ilość znaków: " + text.length() + "</h1>");
+    }
+
+    public static void countCharsWithoutSpaces(String text, PrintWriter writer) {
+        int charsWithoutSpaces = 0;
+        for (int i = 0; i < text.length(); i++) {
+            if (text.charAt(i) != ' ')
+                charsWithoutSpaces++;
+        }
+        writer.println("<h1>Ilość znaków bez spacji: " + charsWithoutSpaces + "</h1>");
+    }
+
+    public static void countWords(String text, PrintWriter writer) {
+        String[] split = text.split(" ");
+        int countWords = split.length;
+        writer.println("<h1>Ilość słów: " + countWords + "</h1>");
+    }
+
+    public static void isPalindrome(String text, PrintWriter writer) {
+        String reversed = "";
```

Comment:
Twój algorytm cierpi na dużę i małe litery. Jak wpiszesz "Ala" który jest palindromem Twój zwróci false. Jest to związane z tym że litera "A" != "a". Potrzebujesz gdzieś wywołać jeszcze `toLowerCase()`

Reply:
Poprawione